### PR TITLE
Add support for rendering app blocks

### PR DESCRIPTION
**Pull request by shime**
Similarly to https://github.com/booqable/kylie-theme/pull/116, this adds support for app blocks to the header and the footer section.

Should not get merged until support for this has landed in the main app.

### CODE REVIEW

**Comment by jevin** [comment]
Looks good!

I'm curious about one thing though. Have you considered moving the app initialisations/includes outside of the theme? Take a look at how `{{ content_for_body }}` works in `layout/theme.liquid`. `content_for_body` is defined in the drops here: `app/lib/shop_next/drops/global_drop.rb:165`

The advantage here is that it makes it easier for theme developers outside of Booqable to setup apps in their themes. They can just call a helper function instead of having to create a snippet and then include it in the header/footer etc.

Let me know if you need more info on this!

**Review by dabrorius — APPROVED** [review]

**Comment by srgzlb** [comment]
Can we have a `render` tag here instead of the deprecated `include` tag?

File: sections/footer.liquid [hunk 1]
```diff
@@ -406,6 +408,9 @@
             "default": "left"
           }
         ]
+      },
+      {
+        "type": "@app"
```
Comment (jevin) [inline sections/footer.liquid:413]: Maybe it's a case of the Monday mornings, but will "@app" match `{% when "app" %}` ?

**Comment by jevin** [comment]
> Can we have a `render` tag here instead of the deprecated `include` tag?

This could also be a security issue because `include` will expose all the variables to the App, whereas render allows us to choose what variables to expose.

File: sections/footer.liquid (hunk 1 above)
Comment (shime) [reply sections/footer.liquid:413]: It will, which is a little surprising.

**Comment by srgzlb** [comment]
> This could also be a security issue because `include` will expose all the variables to the App, whereas render allows us to choose what variables to expose.

Make sense, I missed this moment

**Comment by shime** [comment]
I've just changed it to use the `render` tag instead of the deprecated `include` tag for future proofing. I don't think using the original `include` tag exposes us to security risks as it renders an internal template that's under our control and which uses an isolated renderer specifically designed for app rendering.

**Review by srgzlb — APPROVED** [review]

**Review by zmajstor — APPROVED** [review]

File: sections/footer.liquid [hunk 2]
```diff
@@ -120,17 +121,21 @@
...
-        {% render "app", block: block, section: section %}
       {%- endcase -%}
     {%- endfor -%}
   </nav>
 {%- endcapture -%}
 
 {% comment %} Payments and social component {% endcomment %}
 {%- capture footer_icons -%}
-  {%- if facebook != blank or instagram != blank or twitter != blank or linkedin != blank or pinterest != blank or tiktok != blank or tumblr != blank or snapchat != blank or youtube != blank or vimeo != blank or payment_methods.size > 0-%}
+  {%- if facebook != blank or instagram != blank or twitter != blank or linkedin != blank or pinterest != blank or tiktok != blank or tumblr != blank or snapchat != blank or youtube != blank or vimeo != blank or payment_methods.size > 0 or app != blank -%}
```
Comment (zmajstor) [inline sections/footer.liquid:131]: OMG, is Tumblr still a thing? :)

**Review by zmajstor — APPROVED** [review]

**Review by jevin — APPROVED** [review]

**Pull request merged**